low-level/files/opt/seagate/sspl/setup/sspl_config.py

```diff
@@ -48,9 +48,38 @@ class SSPLConfig:
     def __init__(self):
         """Init method for sspl setup config."""
         self._script_dir = os.path.dirname(os.path.abspath(__file__))
-        self.role = None
-        self.config_index = 'sspl'
-        Conf.load(self.config_index, consts.sspl_config_path)
+        # Load sspl and global configs
+        Conf.load(consts.SSPL_CONFIG_INDEX, consts.sspl_config_path)
+        global_config_url = Conf.get(
+            consts.SSPL_CONFIG_INDEX, "SYSTEM_INFORMATION>global_config_dump_url")
+        Conf.load(consts.GLOBAL_CONFIG_INDEX, global_config_url)
+
+    def validate(self):
+        """Validate config for supported role and product"""
+        # Validate role
+        self.role = Conf.get(consts.GLOBAL_CONFIG_INDEX, 'release>setup')
+        if not self.role:
+            raise SetupError(
+                errno.EINVAL,
+                "%s - validation failure. %s",
+                self.name,
+                "Role not found in SSPL global config dump.")
```

Comment:
Now I modified this statement to `"Role not found in global config copy."`
By mentioning it as copy, it will clear tells where the value exactly coming form. This will be useful to track in case global config has required key but dumped config won't have in worst situation.

Reply:
Done.

Reply:
sure